crates/ra_hir/src/ty/method_resolution.rs

```diff
@@ -255,6 +255,26 @@ fn iterate_inherent_methods<T>(
     None
 }
 
+pub(crate) fn implements_trait(
+    ty: &Canonical<Ty>,
+    db: &impl HirDatabase,
+    resolver: &Resolver,
+    krate: Crate,
+    trait_: Trait,
+) -> bool {
+    let env = lower::trait_env(db, resolver);
+    let goal = generic_implements_goal(db, env.clone(), trait_, ty.clone());
+    let solution = db.trait_solve(krate, goal);
+
+    if let Some(solution) = solution {
+        if let Solution::Unique(_) = solution {
```

Comment:
requiring a unique solution is too much -- for completion purposes, we should answer 'yes' if there is some solution at all.